lib/whitehall/document_filter/advanced_search_result.rb

```diff
@@ -52,21 +52,7 @@ def operational_field
   private
 
     def fetch_from_cache(type, slug)
-      Rails.cache.fetch("#{type}-#{slug}", namespace: "results", expires_in: 30.minutes, race_condition_ttl: 1.second) do
-        case type
-        when :organisation
-          Organisation.includes(:translations).find_by(slug: slug)
-        when :topic
-          Classification.find_by(slug: slug)
-        when :document_collection
-          # Don't fall over if index refers a document which has had it's slug changed.
-          Document.find_by(slug: slug).try(:published_edition)
-        when :operational_field
-          OperationalField.find_by(slug: slug)
-        else
-          raise "Can't fetch '#{type}' -- unknown type"
-        end
-      end
+      FetchFromCacheService.new(type, slug).fetch
```

Comment:
I reckon rather than having this as a method you could just replace the method call with the service call.

Reply:
I considered doing that, but thought the lines looked a bit long.

Reply:
The FetchFromCacheService has been removed (as though it never existed) as we have instead removed the `SearchResult` presenter and replaced it with an updated `RummagerDocumentPresenter`.